### ISSUE TITLE
Command history per folder

### ISSUE THREAD

**Issue by AugustinGrigorov** [warpdotdev/Warp#2932]
### Describe the solution you'd like?

```text
I work on multiple repos, and there are some common terminal commands I need to run in them - like build test. Would be great if Warp kept command history per directory, so when I go over commands I ran I'd find the ones for the appropriate repo.
```

### Is your feature request related to a problem? Please describe.

```text
I go to a repo and need to look up what commands I need to run to run the tests or a build.
```

### How important is this feature to you?

3

### Warp Internal (ignore) - linear-label:770f6576-d6c0-4e4f-a259-fc64b5156087

None

**Comment by dannyneira** [comment]
Hi @AugustinGrigorov We appreciate you submitting this request. AFAIK, we have plans to make autosuggestions smarter and provide suggestions based on the folder you're in and common commands you run in said folder (for example).

As a stopgap feature, you could try and use repository Workflows, that way common commands run in a certain repo/folder will stay organized and only available in said folder. https://docs.warp.dev/features/entry/workflows#adding-a-repository-workflow

Please :+1:  and subscribe to this request as it helps us gauge interest. We will post any updates here.

**Comment by hugows** [comment]
Yes!!!1, even more so because we can see the path on the command history data:

![CleanShot 2024-09-14 at 18 26 16@2x](https://github.com/user-attachments/assets/d765e23a-9eb7-4323-9c14-28a00fd3dacc)
